app/assets/javascripts/discourse/app/components/glimmer-header.gjs

```diff
@@ -0,0 +1,240 @@
+
+  appEventsListeners = modifier(() => {
+    this.appEvents.on(
+      "header:keyboard-trigger",
+      this,
+      this.headerKeyboardTrigger
+    );
+    return () => {
+      this.appEvents.off(
+        "header:keyboard-trigger",
+        this,
+        this.headerKeyboardTrigger
+      );
+    };
+  });
+
+  get inTopicRoute() {
+    return this.search.inTopicContext;
+  }
+
+  get customHeaderClasses() {
+    return _customHeaderClasses.join(" ");
+  }
+
+  @action
+  headerKeyboardTrigger(msg) {
+    switch (msg.type) {
+      case "search":
+        this.toggleSearchMenu();
+        break;
+      case "user":
+        this.toggleUserMenu();
+        break;
+      case "hamburger":
+        this.toggleHamburger();
+        break;
+      case "page-search":
+        if (!this.togglePageSearch()) {
+          msg.event.preventDefault();
+          msg.event.stopPropagation();
+        }
+        break;
+    }
+  }
+
+  @action
+  toggleSearchMenu() {
+    if (this.site.mobileView) {
+      const context = this.search.searchContext;
+      let params = "";
+      if (context) {
+        params = `?context=${context.type}&context_id=${context.id}&skip_context=${this.skipSearchContext}`;
+      }
+
+      if (this.router.currentRouteName === "full-page-search") {
+        scrollTop();
+        document.querySelector(".full-page-search").focus();
+        return false;
+      } else {
+        return DiscourseURL.routeTo("/search" + params);
+      }
+    }
+
+    this.search.visible = !this.search.visible;
+    if (!this.search.visible) {
+      this.search.highlightTerm = "";
+      this.search.inTopicContext = false;
+      document.getElementById(SEARCH_BUTTON_ID)?.focus();
+    }
+  }
+
+  @action
+  togglePageSearch() {
+    this.search.inTopicContext = false;
+
+    let showSearch = this.router.currentRouteName.startsWith("topic.");
+    // If we're viewing a topic, only intercept search if there are cloaked posts
+    if (showSearch) {
+      const controller = this.register.lookup("controller:topic");
+      const total = controller.get("model.postStream.stream.length") || 0;
+      const chunkSize = controller.get("model.chunk_size") || 0;
+      showSearch =
+        total > chunkSize &&
+        document.querySelector(
+          ".topic-post .cooked, .small-action:not(.time-gap)"
+        )?.length < total;
+    }
+
+    if (this.search.visible) {
+      this.toggleSearchMenu();
+      return showSearch;
+    }
+
+    if (showSearch) {
+      this.search.inTopicContext = true;
+      this.toggleSearchMenu();
+      return false;
+    }
+
+    return true;
+  }
+
+  @action
+  toggleUserMenu() {
+    this.header.userVisible = !this.header.userVisible;
+    this.toggleBodyScrolling(this.header.userVisible);
+    this.args.animateMenu();
+
+    // auto focus on first button in dropdown
+    schedule("afterRender", () =>
+      document.querySelector(".user-menu button")?.focus()
+    );
```

Comment:
Could we remove this and put the `{{auto-focus}}` modifier on the appropriate element of the user menu?